functional/FunctionalTest.java

```diff
@@ -3633,66 +3621,49 @@ public static void getDefaultRetention_test() throws Exception {
+        client.setDefaultRetention(
+            SetDefaultRetentionArgs.builder().bucket(bucketName).config(expectedConfig).build());
+        ObjectLockConfiguration config =
+            client.getDefaultRetention(
+                GetDefaultRetentionArgs.builder().bucket(bucketName).build());
 
-        if ((!(config.duration().unit() == expectedConfig.duration().unit()
-                && config.duration().duration() == expectedConfig.duration().duration()))
-            || (config.mode() != expectedConfig.mode())) {
+        if (config.mode() != expectedConfig.mode()) {
           throw new Exception(
-              "[FAILED] Expected: expected duration : "
-                  + expectedConfig.duration()
-                  + ", got: "
-                  + config.duration()
-                  + " expected mode :"
-                  + expectedConfig.mode()
-                  + ", got: "
-                  + config.mode());
+              "[FAILED] mode: expected: " + expectedConfig.mode() + ", got: " + config.mode());
+        }
+
+        if (config.duration().unit() != expectedConfig.duration().unit()
+            || config.duration().duration() != expectedConfig.duration().duration()) {
+          throw new Exception(
+              "[FAILED] duration: " + expectedConfig.duration() + ", got: " + config.duration());
         }
 
         expectedConfig =
             new ObjectLockConfiguration(RetentionMode.GOVERNANCE, new RetentionDurationYears(1));
-        client.setDefaultRetention(bucketName, expectedConfig);
-        config = client.getDefaultRetention(bucketName);
+        client.setDefaultRetention(
+            SetDefaultRetentionArgs.builder().bucket(bucketName).config(expectedConfig).build());
+        config =
+            client.getDefaultRetention(
+                GetDefaultRetentionArgs.builder().bucket(bucketName).build());
 
-        if ((!(config.duration().unit() == expectedConfig.duration().unit()
-                && config.duration().duration() == expectedConfig.duration().duration()))
-            || (config.mode() != expectedConfig.mode())) {
+        if (config.mode() != expectedConfig.mode()) {
           throw new Exception(
-              "[FAILED] Expected: expected duration : "
-                  + expectedConfig.duration()
-                  + ", got: "
-                  + config.duration()
-                  + " expected mode :"
-                  + expectedConfig.mode()
-                  + ", got: "
-                  + config.mode());
+              "[FAILED] mode: expected: " + expectedConfig.mode() + ", got: " + config.mode());
         }
+
+        if (config.duration().unit() != expectedConfig.duration().unit()
+            || config.duration().duration() != expectedConfig.duration().duration()) {
+          throw new Exception(
+              "[FAILED] duration: " + expectedConfig.duration() + ", got: " + config.duration());
+        }
```

Comment:
This whole code seems to be same as the previous lines except for the retention mode and duration. It can be made into a method and called twice with different arguments. Something like:

```
  testGetDefaultRetentionFor(RetentionMode.COMPLIANCE, new RetentionDurationDays(10));
  testGetDefaultRetentionFor(RetentionMode.GOVERNANCE, new RetentionDurationYears(1));
```

Reply:
done